File: app/src/main/java/fi/tuska/jalkametri/util/AssertionUtils.java
package fi.tuska.jalkametri.util;

public final class AssertionUtils {

    public static boolean isAssertionsEnabled() {
        try {
            // This should cause an AssertionError to be thrown
            assert false;
        } catch (AssertionError e) {
            // AssertionError thrown
            return true;
        }
        // Not thrown, so assertions are not on
        return false;
    }

}

File: app/src/main/java/fi/tuska/jalkametri/util/AssertionUtils.kt
package fi.tuska.jalkametri.util

import fi.tuska.jalkametri.PrivateData

object AssertionUtils {

    val TAG = "AssertionUtils"

    val isAssertionsEnabled: Boolean = PrivateData.ASSERTIONS_ENABLED

    fun <T> expectEquals(v1: T, v2: T) {
        if (v1 != v2) {
            val e = AssertionError("$v1 != $v2")
            LogUtil.e(TAG, "Assertion failed: ${e.message}: ${e.stackTrace}")
            if (isAssertionsEnabled) {
                throw e
            }
        }
    }

    inline fun expect(cond: () -> Boolean) {
        if (!cond()) {
            val e = AssertionError()
            LogUtil.e(TAG, "Assertion failed: ${e.stackTrace}")
            if (isAssertionsEnabled) {
                throw e
            }
        }
    }

    fun expect(cond: Boolean) {
        if (!cond) {
            val e = AssertionError()
            LogUtil.e(TAG, "Assertion failed: ${e.stackTrace}")
            if (isAssertionsEnabled) {
                throw e
            }
        }
    }

}
